client/src/app/site/projector/services/current-list-of-speakers.service.ts

```diff
@@ -2,13 +2,39 @@ import { Injectable } from '@angular/core';
 
 import { BehaviorSubject, Observable, Subscription } from 'rxjs';
 
+import { Follow } from 'app/core/core-services/model-request-builder.service';
 import { ListOfSpeakersRepositoryService } from 'app/core/repositories/agenda/list-of-speakers-repository.service';
 import { ProjectorRepositoryService } from 'app/core/repositories/projector/projector-repository.service';
-import { ProjectorService } from 'app/core/ui-services/projector.service';
-import { ViewListOfSpeakers } from 'app/site/agenda/models/view-list-of-speakers';
-import { SlideManager } from 'app/slides/services/slide-manager.service';
+import { hasListOfSpeakers, ViewListOfSpeakers } from 'app/site/agenda/models/view-list-of-speakers';
 import { ViewProjector } from '../models/view-projector';
 
+/**
+ * A `Follow`-object to request the reference projector of a meeting
+ * and depending on it the content-object's list of speakers.
+ */
+export const CURRENT_LIST_OF_SPEAKERS_FOLLOW: Follow = {
+    idField: 'reference_projector_id',
+    follow: [
+        {
+            idField: 'current_projection_ids',
+            follow: [
+                {
+                    idField: 'content_object_id',
+                    follow: [
+                        {
+                            idField: 'list_of_speakers_id',
+                            follow: [
+                                { idField: 'speaker_ids', follow: [{ idField: 'user_id', fieldset: 'shortName' }] }
+                            ]
+                        }
+                    ]
+                }
+            ],
+            fieldset: 'content'
+        }
+    ]
+};
```

Comment:
I don't know, if we achieve this in a smarter way...

Reply:
No, this is right - you request it exactly as the requirements say: "the autopilot shows the username of the list of speakers of the currently projected slide on the reference projector". It just resembles this perfectly.